src/NugetReferenceResolver/RuntimeGraph.cs

```diff
@@ -0,0 +1,926 @@
+        }
+
+        private static IEnumerable<RuntimeDefinition> CreateRuntimeDefinitions(
+            IDictionary<string, JToken> definition,
+            IDictionary<string, JToken> runtimes)
+        {
+            if (!definition.ContainsKey("#import"))
+            {
+                return Enumerable.Empty<RuntimeDefinition>();
+            }
+            else
+            {
+                var fallbacks = (JArray)definition["#import"];
+                return fallbacks
+                    .Select(fr => (string)fr)
+                    .Select(fr => {
+                        if (runtimes.FirstOrDefault(r => r.Key.Equals(fr)).Key == null)
+                        {
+                            return new RuntimeDefinition
+                            {
+                                Name = fr,
+                                Fallbacks = Enumerable.Empty<RuntimeDefinition>()
+                            };
+                        }
+                        return new RuntimeDefinition
+                        {
+                            Name = runtimes.First(r => r.Key.Equals(fr)).Key,
+                            Fallbacks = CreateRuntimeDefinitions(
+                                (IDictionary<string, JToken>)runtimes.First(r => r.Key.Equals(fr)).Value,
+                                runtimes).ToArray()
+                        };
+                    });
+            }
+        }
+
+        public RuntimeGraph(IEnumerable<RuntimeDefinition> runtimes)
+        {
+            Runtimes = runtimes;
+        }
+
+        public IEnumerable<RuntimeDefinition> Runtimes { get; set; }
+
+        public static IEnumerable<string> GetCompatibleRuntimes(string runtimeId)
+        {
+            var runtimes = new HashSet<string>();
+            var runtime = Instance.Runtimes.FirstOrDefault(r => r.Name.Equals(runtimeId));
+            if (runtime != null)
+            {
+                var pendingRuntimes = new Stack<RuntimeDefinition>();
+                pendingRuntimes.Push(runtime);
+                while (pendingRuntimes.Count > 0)
+                {
+                    var currentRuntime = pendingRuntimes.Pop();
+                    runtimes.Add(currentRuntime.Name);
+                    foreach (var fallback in currentRuntime.Fallbacks)
+                    {
+                        pendingRuntimes.Push(fallback);
+                    }
+                }
+            }
+
+            return runtimes;
+        }
+
+        public static string GetCurrentRuntimeId()
+        {
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
+            {
+                return RuntimeEnvironment.GetRuntimeIdentifier();
+            }
+            else
+            {
+                var arch = RuntimeEnvironment.RuntimeArchitecture.ToLowerInvariant();
+                if (RuntimeEnvironment.OperatingSystemVersion.StartsWith("6.1", StringComparison.Ordinal))
```

Comment:
Why have you re-implemented what is already returned by `RuntimeEnvironment.GetRuntimeIdentifier()`?